test/common/network/matching/inputs_integration_test.cc

```diff
@@ -171,6 +208,25 @@ TEST_F(InputsIntegrationTest, ApplicationProtocolInput) {
   EXPECT_TRUE(result.on_match_.has_value());
 }
 
+TEST_F(InputsIntegrationTest, FilterStateInput) {
+  std::string key = "filter_state_key";
+  std::string value = "filter_state_value";
+  initializeFilterStateCase("FilterStateInput", key, value);
+
+  StreamInfo::FilterStateImpl filter_state(StreamInfo::FilterState::LifeSpan::FilterChain);
+  filter_state.setData(key, std::make_shared<Router::StringAccessorImpl>(value),
+                       StreamInfo::FilterState::StateType::Mutable,
+                       StreamInfo::FilterState::LifeSpan::Connection,
+                       StreamInfo::FilterState::StreamSharing::SharedWithUpstreamConnection);
+
+  Network::MockConnectionSocket socket;
+  MatchingDataImpl data(socket, filter_state);
+
+  const auto result = match_tree_()->match(data);
+  EXPECT_EQ(result.match_state_, Matcher::MatchState::MatchComplete);
+  EXPECT_TRUE(result.on_match_.has_value());
```

Comment:
Can you add a negative case (e.g. when the state is missing or the key doesn't match)?

Reply:
done